src/neva/pal_service/agl/BUILD.gn

```diff
@@ -15,10 +15,37 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import("//build/config/features.gni")
+import("//build/config/linux/pkg_config.gni")
 import("//neva/pal_service/pal_service.gni")
+import("//third_party/protobuf/proto_library.gni")
 
 assert(use_dbus)
 
+pkg_config("grpc_libs") {
+  packages = [
+    "protobuf",
+    "grpc",
+    "grpc++",
+  ]
+
+  defines = [
+    "GRPC_USE_PROTO_LITE",
+    "HAVE_CONFIG_H",
+    "PB_FIELD_16BIT",
+    "GRPC_NO_XDS",
+    "GRPC_NO_RLS",
+  ]
+}
```

Comment:
I think, in Chromium codebase, you should add //third_party/grpc:grpc++ to dependencies, instead of detecting library though pkg_config. This is because grpc++ is part of Chromium distribution already. Same about //third_party/grpc:grpc_config (see other examples in codebase to see how grpc is used in Chromium usually).

Reply:
That was my first attempt, but during the process of creating a new release for webos, they remove several third party libraries and [grpc was one of them](https://github.com/rogerzanoni/chromium91/blob/%4017.agl/scripts/build/remove-nonessential-files.sh#L134). That I did at first was to checkout the correct grpc revision on third party and make a commit adding it back, and used grpc_library() instead of proto_library, but I started having issues with yocto because more than one recipe was installing libgrpc. I did it with proto_library and changed the protoc  wrapper code to allow an external plugin to avoid that, but I could also go back to the approach of adding back third_party/grpc and linking it statically with libcbe to avoid issues with recipes that use grpc from the oe devtools. Wdyt?

Reply:
That's interesting. I guess it is related to the fact that WAM is using grpc too? It looks like the right solution would be moving to use third_party one, and export the symbols so wam can use it too, but it looks too much complicated.

I'd say we would want to have in //third_party/grpc BUILD.gn this configuration still, so it can use system library instead of third_party, and have a GN arg for that purpose. Something like use_system_libgrpc.

Reply:
Yes, the problem was because wam uses grpc too, but it would conflict with other recipes, like kuksa or the other home demo apps. I like the use_system_libgrpc approach, maybe I can just add [grpc_library.gni](https://source.chromium.org/chromium/chromium/src/+/main:third_party/grpc/grpc_library.gni;l=8?q=grpc_libra&sq=&ss=chromium%2Fchromium%2Fsrc) back, and use this flag to decide which extra_configs and which generator to use. But I would need to keep my changes to proto_library and the wrapper, because proto_library expects invoker.generator_plugin_label to one of the plugins built with chromium and doesn't allow extenal plugins.

Reply:
I ended up reverting to my previous approach, now I have one commit adding back third_party/grpc and other changing the appservice_delegate. Previously I missed the grpc_use_static_linking arg that I added to chromium recipe. It's adding a lot of code (grpc + grpc submodules) but it's less hacky now.

Reply:
Forgot to mention, but I changed a bit the JSON that is returned on the application list, now it returns all the attributes instead of just the app ids, I made two follow-up changes on html5-homescreen and launcher.